src/PowerShellEditorServices/Services/Analysis/AnalysisService.cs

```diff
@@ -50,250 +52,88 @@ public class AnalysisService : IDisposable
-
         private readonly ConfigurationService _configurationService;
-
         private readonly ConcurrentDictionary<string, (SemaphoreSlim, Dictionary<string, MarkerCorrection>)> _mostRecentCorrectionsByFile;
 
-        #endregion // Private Fields
+        private CancellationTokenSource _existingRequestCancellation;
+        private readonly SemaphoreSlim _existingRequestCancellationLock;
 
-        #region Properties
+        #endregion
 
-        /// <summary>
-        /// Set of PSScriptAnalyzer rules used for analysis.
-        /// </summary>
-        public string[] ActiveRules { get; set; }
+        #region Properties
 
         /// <summary>
         /// Gets or sets the path to a settings file (.psd1)
         /// containing PSScriptAnalyzer settings.
         /// </summary>
-        public string SettingsPath { get; set; }
+        public string SettingsPath { get; internal set; }
 
         #endregion
 
         #region Constructors
 
-        /// <summary>
-        /// Construct a new AnalysisService object.
-        /// </summary>
-        /// <param name="analysisRunspacePool">
-        /// The runspace pool with PSScriptAnalyzer module loaded that will handle
-        /// analysis tasks.
-        /// </param>
-        /// <param name="pssaSettingsPath">
-        /// The path to the PSScriptAnalyzer settings file to handle analysis settings.
-        /// </param>
-        /// <param name="activeRules">An array of rules to be used for analysis.</param>
-        /// <param name="logger">Maintains logs for the analysis service.</param>
-        /// <param name="pssaModuleInfo">
-        /// Optional module info of the loaded PSScriptAnalyzer module. If not provided,
-        /// the analysis service will populate it, but it can be given here to save time.
-        /// </param>
-        private AnalysisService(
-            RunspacePool analysisRunspacePool,
-            string pssaSettingsPath,
-            IEnumerable<string> activeRules,
-            ILanguageServer languageServer,
-            ConfigurationService configurationService,
-            ILogger logger,
-            PSModuleInfo pssaModuleInfo = null)
+        public AnalysisService(ConfigurationService configurationService, ILanguageServer languageServer, ILoggerFactory factory)
         {
-            _analysisRunspacePool = analysisRunspacePool;
-            SettingsPath = pssaSettingsPath;
-            ActiveRules = activeRules.ToArray();
-            _languageServer = languageServer;
+            SettingsPath = configurationService.CurrentSettings.ScriptAnalysis.SettingsPath;
+            _logger = factory.CreateLogger<AnalysisService>();
+            _analyzer = new HostedAnalyzer();
+            _analyzerSettings = _analyzer.CreateSettings(s_includedRules);
+            _analyzerSettings.Severities.AddRange(new [] {
+                RuleSeverity.Error.ToString(),
+                RuleSeverity.Information.ToString(),
```

Comment:
Should this be `RuleSeverity.Warning`? Why is it that these are strings if we've gone to the effort to load the types?